src/lib/writers/OutputFileWriter.ts

```diff
@@ -0,0 +1,44 @@
+import * as fs from 'node:fs';
+import {OutputFormat, RunResults} from '@salesforce/code-analyzer-core';
+import {BundleName, getMessage} from '../messages';
+
+export interface OutputFileWriter {
+	writeToFiles(results: RunResults): void;
+	getOutputFiles(): string[];
+}
+
+export class OutputFileWriterImpl implements OutputFileWriter {
+	private outputFilesToFileFormat: Map<string, OutputFormat> = new Map();
+
+	public constructor(files: string[]) {
+		for (const file of files) {
+			if (file.endsWith(`.csv`)) {
+				this.outputFilesToFileFormat.set(file, OutputFormat.CSV);
+			} else if (file.endsWith('.html') || file.endsWith('.htm')) {
+				throw new Error('TODO: Support HTML-type output');
+			} else if (file.endsWith('.sarif') || file.endsWith('.sarif.json')) {
+				throw new Error('TODO: Support SARIF-type output');
+			// Check for `.json` AFTER checking for `.sarif.json`!
+			} else if (file.endsWith('.json')) {
+				this.outputFilesToFileFormat.set(file, OutputFormat.JSON);
+			} else if (file.endsWith('.junit') || file.endsWith('.junit.xml')) {
+				throw new Error('TODO: Support JUNIT-type output');
+			// Check for `.xml` AFTER checking for `.junit.xml`!
+			} else if (file.endsWith('.xml')) {
+				this.outputFilesToFileFormat.set(file, OutputFormat.XML);
+			} else {
+				throw new Error(getMessage(BundleName.OutputFileWriter, 'error.unrecognized-file-format', [file]));
+			}
+		}
+	}
+
+	public getOutputFiles(): string[] {
```

Comment:
The only place this is used is `run.test.ts`, because I didn't want to do the `.create()` pattern just so I could spy on what files were being passed in. I'm willing to do that replacement if people prefer that, though.

Reply:
It's used in the `--output-file` tests starting at line 232. Those tests are meant to assert that the files passed into `run` are ultimately passed to the OutputFileWriter, which is why I need access to that information.
One alternative workaround would be to add an `addOutfiles()` method that I could spy on.